core/src/main/scala/org/apache/spark/util/ShutdownHookManager.scala

```diff
@@ -209,9 +209,7 @@ private [util] class SparkShutdownHookManager {
 private class SparkShutdownHook(private val priority: Int, hook: () => Unit)
   extends Comparable[SparkShutdownHook] {
 
-  override def compareTo(other: SparkShutdownHook): Int = {
-    other.priority - priority
```

Comment:
Do you mean the overflow cause out of order?

But isn't `SparkShutdownHookManager` is an internal class? Where do we use `MaxValue`/`MinValue` and cause the problem?

Reply:
yes, it caused it. I'm not sure about the usage.

Reply:
Yeah, I don't think this affects any external case.